src/RulesetEngine/Context.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SensioLabs\Deptrac\RulesetEngine;
+
+use SensioLabs\Deptrac\Dependency\DependencyInterface;
+
+final class Context
```

Comment:
can you make the context immutable again?

Reply:
done